src/spritekit.cs

```diff
@@ -144,14 +144,17 @@ interface SK3DNode {
 
 	[DisableDefaultCtor] // DesignatedInitializer below
 #if MONOMAC
+	[NoiOS][NoTV][NoWatch][NoMacCatalyst]
```

Comment:
So do you mean something like this:
`[Mac (10,9)][Watch (3,0)][iOS (7,0)][MacCatalyst (13,1)]`
and removing all the attributes from the 3 branches?

Reply:
Correct

Reply:
@rolfbjarne question about this since I have seen other partially similar things/potential issues. What exactly in this situation makes us have to move the attributes outside the branches?

Reply:
> What exactly in this situation makes us have to move the attributes outside the branches?

If the member inside both branches is the same member.

Example:

```cs
#if MONOMAC
    [NoiOS]
    [Mac(12,0)]
    void Foo ();
#else
    [iOS(14,0)]
    [NoMac]
    void Foo ();
#endif
```

If you try to use the method `Foo` on macOS, intellisense will say it's not available on iOS. Likewise, if you try to use the method on iOS, intellisense will say it's not available on macOS.

Both are lies, since the method is available on both platforms. This is the right way:

```cs
    [Mac(12,0)]
    [iOS(14,0)]
    void Foo ();
```

Now things become a bit fuzzy if the members aren't *exactly* the same, for instance:


```cs
#if MONOMAC
    [NoiOS]
    [Mac(12,0)]
    void Foo (NSImage image);
#else
    [iOS(14,0)]
    [NoMac]
    void Foo (UIImage image);
#endif
```

are these two members the same? I'd argue yes, because the purpose (and most likely the behavior) of these two methods are identical. Another way to determine if two members are identical is to check the selector: if the selectors are identical, it's more than likely that we should consider them equal members in C# as well.

Reply:
@rolfbjarne  Thanks so much for this explanation!!